util/encoding/encoding_test.go

```diff
@@ -264,3 +266,53 @@ func TestEncodeDecodeVarUint64Decreasing(t *testing.T) {
 	}
 	testCustomEncode64(testCases, EncodeVarUint64Decreasing, t)
 }
+
+func TestEncodeDecodeBytes(t *testing.T) {
+	testCases := []struct {
```

Comment:
Add some tests that include 1) multiple occurrences of `\x00` (both consecutive and separated) so the string is built up in multiple segments 2) the special sequences `\x00\x01` and `\x00\xff` in the input data 3) `\x00` at the start of the input.